shunting-yard.h

```diff
@@ -280,9 +281,9 @@ class RefToken : public TokenBase {
   packToken key;
   packToken origin;
   RefToken(packToken k, TokenBase* v, packToken m = packToken::None()) :
-    TokenBase(v->type | REF), original_value(v), key(k), origin(m) {}
+    TokenBase(v->type | REF), original_value(v), key(std::forward<packToken>(k)), origin(std::forward<packToken>(m)) {}
   RefToken(packToken k = packToken::None(), packToken v = packToken::None(), packToken m = packToken::None()) :
-    TokenBase(v->type | REF), original_value(v), key(k), origin(m) {}
+    TokenBase(v->type | REF), original_value(std::forward<packToken>(v)), key(std::forward<packToken>(k)), origin(std::forward<packToken>(m)) {}
```

Comment:
I might be wrong, but in this case a call to `std::forward<packToken>()` wouldn't be equivalent to a call to `std::move`?

It seems that the usual use-case for `std::forward<T>` would be on templates where T might be either `T`, `T&` or `T&&`, in case it is `T&` it returns an lvalue, else (if it is `T` or `T&&`) `std::forward<T>` returns an rvalue just like `std::move` i.e.:

```C++
#include <iostream>

void overloaded( int const &arg ) { std::cout << "by lvalue\n"; }
void overloaded( int && arg ) { std::cout << "by rvalue\n"; }
 
int main() {
    int x = 5;
    
    // Tests with literals, references and forward references:
    overloaded(std::forward<int>(x));
    overloaded(std::forward<int&>(x));
    overloaded(std::forward<int&&>(x));
}
```

Also hosted on REPL for convenience: https://repl.it/@vingarcia/MediumpurpleWeightyWebmaster